ReactAndroid/src/main/java/com/facebook/react/modules/websocket/WebSocketModule.java

```diff
@@ -187,4 +200,31 @@ private void notifyWebSocketFailed(int id, String message) {
     params.putString("message", message);
     sendEvent("websocketFailed", params);
   }
+
+  private static String setDefaultOrigin(String url) {
+    try {
+      String defaultOrigin;
+      String scheme = "";
+
+      URI requestURI = new URI(url);
+      if (requestURI.getScheme().equals("wss")) {
+        scheme += "https";
+      } else if (requestURI.getScheme().equals("ws")) {
+        scheme += "http";
+      }
+
+      if (requestURI.getPort() != -1) {
+        defaultOrigin = String.format("%s://%s:%s", scheme, requestURI.getHost(), requestURI.getPort());
+      } else {
+        defaultOrigin = String.format("%s://%s/", scheme, requestURI.getHost());
+      }
+
+      return defaultOrigin;
+
+    } catch(URISyntaxException e) {
+      // Handle error
```

Comment:
@christopherdro Since this is a user error, we should throw the error to show a redbox (`IllegalArgumentException`). This matches the behaviour on the web where it throws an error.